src/ophys_etl/types/_roi.py

```diff
@@ -286,3 +286,114 @@ def contour_mask(self) -> np.ndarray:
         if self._contour_mask is None:
             self._construct_contour_mask()
         return np.copy(self._contour_mask)
+
+    def get_bounding_box_cutout(self, image: np.ndarray) -> np.ndarray:
+        """Return a cutout from an image that matches the roi bounding box.
+        """
+        return image[self._y0:self._y0 + self._height,
+                     self._x0:self._x0 + self._width]
+
+    def get_centered_cutout(self,
+                            image: np.ndarray,
+                            height: int,
+                            width: int) -> np.ndarray:
+        """Get a cutout of arbitrary size centered on the bounding box
+        centroid.
+
+        Pad cutout with zeros if requested cutout extends outside of image.
+
+        Parameters
+        ----------
+        image : numpy.ndarray, (N, M)
+            Image to create cutout/thumbnail from.
+        height : int
+            Height(y) of output cutout image.
+        width : int
+            Width(x) of output cutout image.
+
+        Returns
+        -------
+        cutout : numpy.ndarray
+            Cutout of requested size centered on the bounding box center.
+        """
+        # Find the indices of the desired cutout in the image.
+        row_indices = self._get_cutout_indices(self.bounding_box_center_y,
+                                               image.shape[-2],
+                                               height)
+        col_indices = self._get_cutout_indices(self.bounding_box_center_x,
+                                               image.shape[-1],
+                                               width)
+        # Get initial cutout.
+        thumbnail = image[row_indices[0]:row_indices[1],
+                          col_indices[0]:col_indices[1]]
+        # Find if we need to pad the image.
+        row_pad = self._get_padding(self.bounding_box_center_y,
+                                    image.shape[-2],
+                                    height)
+        col_pad = self._get_padding(self.bounding_box_center_x,
+                                    image.shape[-1],
+                                    width)
+        # Pad the cutout if needed.
+        padding = (row_pad, col_pad)
+        return np.pad(thumbnail,
+                      pad_width=padding, mode="constant",
+                      constant_values=0)
+
+    @staticmethod
+    def _get_cutout_indices(
```

Comment:
Given that `_get_cutout_indices` and `_get_padding` do not actually use `self`, what do you think about moving them to `src/ophys_etl/utils/array_utils.py` so that they can be used more generally in the future?

Reply:
Sure. I can move the code and tests into array_utils.